server/channels/app/custom_profile_attributes_test.go

```diff
@@ -0,0 +1,462 @@
+		field1 := &model.PropertyField{
+			GroupID: cpaGroupID,
+			Name:    "Field 1",
+			Type:    model.PropertyFieldTypeText,
+		}
+
+		_, err := th.App.Srv().propertyService.CreatePropertyField(field1)
+		require.NoError(t, err)
+
+		field2 := &model.PropertyField{
+			GroupID: model.NewId(),
+			Name:    "Field 2",
+			Type:    model.PropertyFieldTypeText,
+		}
+		_, err = th.App.Srv().propertyService.CreatePropertyField(field2)
+		require.NoError(t, err)
+
+		field3 := &model.PropertyField{
+			GroupID: cpaGroupID,
+			Name:    "Field 3",
+			Type:    model.PropertyFieldTypeText,
+		}
+		_, err = th.App.Srv().propertyService.CreatePropertyField(field3)
+		require.NoError(t, err)
+
+		fields, appErr := th.App.ListCPAFields()
+		require.Nil(t, appErr)
+		require.Len(t, fields, 2)
+
+		fieldNames := []string{}
+		for _, field := range fields {
+			fieldNames = append(fieldNames, field.Name)
+		}
+		require.ElementsMatch(t, []string{"Field 1", "Field 3"}, fieldNames)
+	})
+}
+
+func TestCreateCPAField(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	if cpaGroupID == "" {
+		_, err := th.App.cpaGroupID()
+		require.NoError(t, err)
+	}
+
+	t.Run("should fail if the field is not valid", func(t *testing.T) {
+		field := &model.PropertyField{Name: model.NewId()}
+
+		createdField, err := th.App.CreateCPAField(field)
+		require.NotNil(t, err)
+		require.Empty(t, createdField)
+	})
+
+	t.Run("should not be able to create a property field for a different feature", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: model.NewId(),
+			Name:    model.NewId(),
+			Type:    model.PropertyFieldTypeText,
+		}
+
+		createdField, appErr := th.App.CreateCPAField(field)
+		require.Nil(t, appErr)
+		require.Equal(t, cpaGroupID, createdField.GroupID)
+	})
+
+	t.Run("should correctly create a CPA field", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: cpaGroupID,
+			Name:    model.NewId(),
+			Type:    model.PropertyFieldTypeText,
+			Attrs:   map[string]any{"visibility": "hidden"},
+		}
+
+		createdField, err := th.App.CreateCPAField(field)
+		require.Nil(t, err)
+		require.NotZero(t, createdField.ID)
+		require.Equal(t, cpaGroupID, createdField.GroupID)
+		require.Equal(t, map[string]any{"visibility": "hidden"}, createdField.Attrs)
+
+		fetchedField, gErr := th.App.Srv().propertyService.GetPropertyField(createdField.ID)
+		require.NoError(t, gErr)
+		require.Equal(t, field.Name, fetchedField.Name)
+		require.NotZero(t, fetchedField.CreateAt)
+		require.Equal(t, fetchedField.CreateAt, fetchedField.UpdateAt)
+	})
+
+	t.Run("should not be able to create CPA fields above the limit", func(t *testing.T) {
+		// we create the rest of the fields required to reach the limit
+		for i := 1; i < CustomProfileAttributesFieldLimit-1; i++ {
```

Comment:
This index is like this because we've created two fields in the tests above, right? Should we try and start from scratch in all tests? So that they don't depend on each other?

Reply:
Fixed this comment and the next on be7fc44856, PTAL